### ISSUE TITLE
Investigate authentication with Azure static web apps

### ISSUE THREAD

**Issue by OmarShehata** [Smithsonian/ForestGEO#53]
We want to figure out how to deploy the app/what backend to use. One option is to publish the frontend as a static app that connects to a backend on Azure. 

One requirement in whatever backend design we choose is that we need to be able to handle user authentication.

This tutorial says it can show you how to hook up your React app to an Azure backend + handle authentication:

https://docs.microsoft.com/en-us/learn/modules/publish-static-web-app-authentication/

It would be good to explore this to see if it will work for us.

Sample code from tutorial; https://github.com/MicrosoftDocs/mslearn-staticwebapp-authentication/tree/main

**Comment by OmarShehata** [comment]
One nice thing about this tutorial is it shows you how you can let users login with GitHub/Google/Twitter etc so we don't have to manage our own database of users, which might simplify things to start.

**Comment by zmagar** [comment]
I did a research of [Azure Static Web Apps](https://docs.microsoft.com/en-us/azure/static-web-apps/overview) to use it for our project app and here is a brief summary of my findings:
The service provides convenient way to host the app and build and deploy the app via tight integration with GitHub, and also provides a workflow, it monitors changes in code and when, for example, a pull request is created, Azure can build and deploy a pre-production version of the app so it is easy to see / compare the app with the main development. The other features, that can be useful for our app, include custom domains and Azure Functions (might be useful to implement data validation, etc).
Speaking about authentication, Azure offers integrated authentication with Azure Active Directory, GitHub, Twitter, also when I deployed my app, I found Google and Facebook authentication available (in preview mode):
![Invitation](https://user-images.githubusercontent.com/17448320/178643201-f8065d1e-4e45-420f-a3bb-4bc386f543e5.png)

And there is a custom authentication available using any provider that support OpenID Connect.

Here is a link to my test app's pre-production page where GitHub authentication can be tested:
https://nice-ground-04fc3671e-3.westus2.1.azurestaticapps.net/

An example of how GitHub actions create a working app with the code that developer tries to merge creating a PR:
![preproduction](https://user-images.githubusercontent.com/17448320/178793925-a673ea9f-5c21-475c-8138-c55ce1421d46.png)

Conclusion: 
I think that Azure Static Web Apps service meets our project's requirements, it can help to greatly simplify the deployment of the app and developers workflow by automation of the deployment process and provides a lot of necessary tools to implement the proposed architecture process: web app with custom domain -> csv data upload -> Azure Functions processing (data validation) -> upload the data to Azure cloud storage.
